python/src/scipp/plot/plot.py

```diff
@@ -32,15 +32,18 @@ def _ndarray_to_variable(ndarray):
     return sc.Variable(dims=dims, values=ndarray)
 
 
-def _input_to_data_array(item, key=None):
+def _input_to_data_array(item, all_keys, key=None):
     """
     Convert an input for the plot function to a DataArray or a dict of
     DataArrays.
     """
     to_plot = {}
     if su.is_dataset(item):
         for name in sorted(item.keys()):
-            to_plot[name] = item[name]
+            if name in all_keys:
```

Comment:
So what are the cases where you could have duplicates? Since I thought we were accepting as input either a Dataset, or a dict of DataArrays, i'm not sure how one could end up with duplicates?

Reply:
ah, is it when you pass in a dict of Datasets? I didn't even know we supported that!

Reply:
I found it by accident when I passed an array of Datasets. Keys in the datasets were the same, then get concatenated into inventory dictionary.

Reply:
Instead of appending `_i` to the name, can you just use the combination of the item name in the parent dict and the name of the item in the current dataset? Then it's more explicit which one corresponds to which?

Reply:
Good suggestion